packages/database-editor/src/components/ImportWizard/pages/TableSelectionPage.tsx

```diff
@@ -0,0 +1,60 @@
+import type { DatabaseEditorContext, DatabaseEditorDBContext, DatabaseTable } from '@axonivy/database-editor-protocol';
+import { Button, Flex, Label } from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useQuery } from '@tanstack/react-query';
+import { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useClient } from '../../../protocol/ClientContextProvider';
+import { genQueryKey } from '../../../query/query-client';
+import { notImplemented } from '../ImportWizard';
+import { TableDropdown } from '../TableDropdown';
+
+export type SelectTablesPageProps = {
+  context: DatabaseEditorContext;
+  selectedDatabase: string;
+  updateSelection: (table: DatabaseTable, add: boolean) => void;
+  selectedTables: Array<DatabaseTable>;
+};
+
+export const SelectTablesPage = ({ context, selectedDatabase, updateSelection, selectedTables }: SelectTablesPageProps) => {
+  const { t } = useTranslation();
+  const client = useClient();
+  const [tableContext, setTableContext] = useState<DatabaseEditorDBContext>({
+    app: context.app,
+    pmv: context.pmv,
+    file: context.file,
+    databaseName: selectedDatabase
+  });
+
+  useEffect(() => {
+    setTableContext(() => {
+      return {
+        app: context.app,
+        pmv: context.pmv,
+        file: context.file,
+        databaseName: selectedDatabase
+      };
+    });
+  }, [selectedDatabase, context]);
```

Comment:
do not understand why you need to hold this inside a state...
Why can't you simply set this as context in the usequery?

Or at least just make a useMemo out of it...